api/auth_middleware.go

```diff
@@ -172,3 +175,107 @@ func (app *ApiServer) getUserIDFromWallet(ctx context.Context, wallet string) (i
 	app.resolveWalletCache.Set(key, userId)
 	return userId, nil
 }
+
+/*
+* Parses query string for a signed comms GET request and returns the userId
+associated with the signing wallet
+*/
+func (app *ApiServer) userIdForSignedCommsRequest(c *fiber.Ctx) (int, error) {
+	if c.Method() != "GET" {
+		return 0, fiber.NewError(fiber.StatusBadRequest, "readSignedGet: bad method: "+c.Method())
+	}
+
+	sigBase64 := c.Get(comms.SigHeader)
+
+	// for websocket request, read from query param instead of header
+	if querySig := c.Query("signature"); sigBase64 == "" && querySig != "" {
+		sigBase64 = querySig
+	}
+
+	// Check that timestamp is not too old
+	timestamp, err := strconv.ParseInt(c.Query("timestamp"), 0, 64)
+	if err != nil {
+		return 0, fiber.NewError(fiber.StatusBadRequest, "failed to parse timestamp: "+err.Error())
+	}
+
+	tsAge := time.Now().UnixMilli() - timestamp
+	if tsAge < 0 {
+		tsAge *= -1
+	}
+	if tsAge > comms.SignatureTimeToLiveMs {
+		return 0, fiber.NewError(fiber.StatusBadRequest, "timestamp not current")
+	}
+
+	// Strip out the app_name and api_key query parameters to get the true signature payload
+	// Get the current request URI and parse it
+	uri := c.Request().URI()
+	path := string(uri.Path())
+	query := string(uri.QueryString())
+
+	// Parse query parameters
+	queryParams := make(map[string]string)
```

Comment:
can we use the url package for this?

Reply:
I will check. This was a conversion from a similar function in the old comms repo. I'm wary of changing it too much because if we miss or modify a field it will result in a different signature 😬